### PR TITLE
Add support for `single-result` and `max-length` fixes #9687

### LINKED ISSUE #9687: Can we allow amp-list to accept object instead of arrays in the json?
Right now `amp-list` is limited to handling responses that return an array somewhere in the json, it then iterates the array populating a template for each element.

Would it be possible to have it also accept non array values?  In the case where the response was not an array it would treat it as a single entry array and populate the template once. This might be as simple as when the target is not an array just wrapping it in one rather than throwing an error. 
 
This would greatly increase the use cases for amp-list with existing API endpoints.

### PR DESCRIPTION
**Pull request by jpettitt**
Adds a `single-result` attribute to allow `amp-list` to render from json endpoints that return non-array values.

Adds a `max-length` attribute to tell `amp-list` to render at most `max-length` entries.

### CODE REVIEW

File: extensions/amp-list/validator-amp-list.protoascii [hunk 1]
```diff
@@ -41,6 +41,8 @@ tags: {  # <amp-list>
     blacklisted_value_regex: "__amp_source_origin"
   }
   attrs: { name: "items" }
+  attrs: { name: "single-item" }
```
Comment (honeybadgerdontcare) [inline extensions/amp-list/validator-amp-list.protoascii]: nit: alphabetize the non-bind attributes. also noticed `items` wasn't in previous version, please alphabetize that too. thx!

File: extensions/amp-list/validator-amp-list.protoascii [hunk 2]
```diff
@@ -31,6 +31,9 @@ tags: {  # <amp-list>
   tag_name: "AMP-LIST"
   requires_extension: "amp-list"
   attrs: { name: "credentials" }
+  attrs: { name: "items" }
+  attrs: { name: "max-length" }
```
Comment (aghassemi) [inline extensions/amp-list/validator-amp-list.protoascii]: as discussed, let's do `max-items`

Comment (jpettitt) [reply extensions/amp-list/validator-amp-list.protoascii]: making the change now

File: extensions/amp-list/0.1/amp-list.js [hunk 3]
```diff
@@ -147,6 +147,18 @@ export class AmpList extends AMP.BaseElement {
...
+      if (this.element.hasAttribute('single-result')) {
+        user().assert(typeof items !== 'undefined' ,
+            'Response must contain an arrary or object at "%s". %s',
+            itemsExpr, this.element);
+        if (!isArray(items)) {
+          items = [items];
+        }
+      }
+      const maxLen = parseInt(this.element.getAttribute('max-items'), 10);
+      if (maxLen < items.length) {
```
Comment (aghassemi) [inline extensions/amp-list/0.1/amp-list.js]: we need to do this after the `user().assert(isArray(items)` below

Comment (jpettitt) [reply extensions/amp-list/0.1/amp-list.js]: done, tests running now

**Comment by honeybadgerdontcare** [comment]
The validator changes are now live in prod.

**Pull request merged**